src/tikzDevice.c

```diff
@@ -172,6 +172,7 @@ SEXP TikZ_StartDevice ( SEXP args ){
   int maxSymbolicColors = asInteger(CAR(args)); args = CDR(args);
   Rboolean timestamp = asLogical(CAR(args)); args = CDR(args);
   Rboolean verbose = asLogical(CAR(args)); args = CDR(args);
+  const char *pictureOptions = CHAR(asChar(CAR(args))); args = CDR(args);
```

Comment:
I'm slightly worried about protection here, because `asChar()` might allocate memory in corner cases.

Reply:
Maybe `CHAR(PROTECT(asChar(...)))`, and then increase the number passed to `UNPROTECT()` by one?